lxd/storage/drivers/driver_ceph.go

```diff
@@ -112,6 +113,29 @@ func (d *ceph) FillConfig() error {
 		d.config["ceph.osd.pg_num"] = "32"
 	}
 
+	if d.config["ceph.osd.pool_size"] == "" {
+		size, err := shared.TryRunCommand("ceph",
```

Comment:
Please can you help me understand the reason we are copying the osd_pool_default_size out of the ceph installation into the pool's config if its not specified. This seems to be setting a new precedent of reading a property of the environment and then persisting it into the pool's config.

Reply:
So there is no difference between ceph using the default value or the LXD supplied value for pool size, so I thought it would be helpful to see the default value that the pool was created with, without having to check ceph directly. 

Otherwise the UX requires a lot of back-and-forth between ceph & LXD:
* Create ceph cluster & osds (using ceph)
* Create osd pool (using LXD)
* Check the osd pool size (using ceph)
* Update the osd pool size (using LXD) 

You're right that it's a bit inconsistent with other fields like `pg_num` though, so I can remove it.

Reply:
so it this setting just stored for posterity and not used again once the pool is created or the setting is updated?

Reply:
If a user updates the LXD config on the storage pool manually, then the change will be propagated to ceph. LXD doesn't communicate with ceph periodically, so if the pool size changes through Ceph, LXD will not see the change. So this first value is effectively just a snapshot of the pool size from when the pool was created.

Reply:
So shall I remove it since it doesn't match the behaviour of other keys?

Reply:
I think its ok to leave as-is, either way its potentially out-of-date with reality, so this way its just recording what has been used which can be useful.